### PR TITLE
docstrings + rST formatting

### PR DESCRIPTION
**Pull request by nvy-0x**

### CODE REVIEW

File: eth_portfolio/portfolio.py [hunk 1]
```diff
@@ -142,39 +309,98 @@ async def describe(self, block: int) -> PortfolioBalances:
...
+    :ivar internal_transfers: A container for all internal transfers to or from any of your :class:`PortfolioAddress`.
+    :vartype internal_transfers: PortfolioInternalTransfersLedger
+    :ivar token_transfers: A container for all token transfers to or from any of your :class:`PortfolioAddress`.
+    :vartype token_transfers: PortfolioTokenTransfersLedger
+    :ivar asynchronous: True if default mode is async, False if sync.
+    :vartype asynchronous: bool
+    """
     def __init__(self, portfolio: "Portfolio") -> None:
+        """
+        Initialize a PortfolioLedger instance.
```
Comment (BobTheBuidler) [inline eth_portfolio/portfolio.py:358]: You should no initialize these yourself. They are initialized internally.

File: eth_portfolio/portfolio.py [hunk 2]
```diff
@@ -73,7 +184,20 @@ def __init__(
...
+        """
+        Initialize a Portfolio instance.
 
+        :param addresses: The addresses to include in the portfolio.
+        :type addresses: Addresses
+        :param start_block: The starting block number, defaults to 0.
+        :type start_block: int
+        :param label: A label for the portfolio, defaults to "your portfolio".
+        :type label: str
+        :param load_prices: Whether to load prices, defaults to True.
```
Comment (BobTheBuidler) [inline eth_portfolio/portfolio.py:196]: Whether to load prices when fetching account history, defaults to True.

File: eth_portfolio/portfolio.py [hunk 3]
```diff
@@ -29,41 +44,137 @@
...
 class PortfolioWallets(Iterable[PortfolioAddress], Dict[Address, PortfolioAddress]):  # type: ignore [misc]
     """
-    A container that holds all `PortfolioAddress` objects for a specific `Portfolio`.
+    A container that holds all :class:`PortfolioAddress` objects for a specific :class:`Portfolio`.
+
     Works like a `Dict[Address, PortfolioAddress]` except when you iterate you get the values instead of the keys.
-    You should not initialize these. They are created automagically during Portfolio init.
+    You should not initialize these. They are created automagically during :class:`Portfolio` init.
+
+    :ivar _wallets: A checksummed dictionary of PortfolioAddress objects.
```
Comment (BobTheBuidler) [inline eth_portfolio/portfolio.py:52]: move the _wallets info to the line immediately below the assignment of the self._wallets var

File: eth_portfolio/portfolio.py [hunk 4]
```diff
@@ -29,41 +44,137 @@
...
+    Used to export information about a group of :class:`PortfolioAddress` objects.
+
+    - Has all attributes of a :class:`PortfolioAddress`.
+    - All calls to `function(*args, **kwargs)` will return `{address: PortfolioAddress(Address).function(*args, **kwargs)}`
+
+    :ivar _start_block: The starting block number.
+    :vartype _start_block: int
+    :ivar label: A label for the portfolio.
+    :vartype label: str
+    :ivar load_prices: Whether to load prices.
```
Comment (BobTheBuidler) [inline eth_portfolio/portfolio.py:168]: move these infos to the line immediately below the assignment of each var

File: eth_portfolio/portfolio.py [hunk 5]
```diff
@@ -1,3 +1,18 @@
+"""
+This module defines the `Portfolio` and `PortfolioWallets` classes, which are used to manage and interact with a collection of `PortfolioAddress` objects.
+It also includes the `PortfolioLedger` class for handling transactions, internal transfers, and token transfers associated with the portfolio.
+The `Portfolio` class leverages asynchronous operations to efficiently gather data and perform various portfolio-related tasks.
+
+Dependencies:
```
Comment (BobTheBuidler) [inline eth_portfolio/portfolio.py:6]: rip this out, dependencies dont need mention here

File: eth_portfolio/portfolio.py [hunk 6]
```diff
@@ -1,3 +1,18 @@
+"""
+This module defines the `Portfolio` and `PortfolioWallets` classes, which are used to manage and interact with a collection of `PortfolioAddress` objects.
+It also includes the `PortfolioLedger` class for handling transactions, internal transfers, and token transfers associated with the portfolio.
+The `Portfolio` class leverages asynchronous operations to efficiently gather data and perform various portfolio-related tasks.
```
Comment (BobTheBuidler) [inline eth_portfolio/portfolio.py:4]: change this so it references how a_sync impacts the behavior of the Portfolio object. ping me if you dont know what this means

File: eth_portfolio/portfolio.py [hunk 7]
```diff
@@ -29,41 +36,124 @@
...
+        :param portfolio: The :class:`Portfolio` instance to which this wallet belongs.
+        :type portfolio: Portfolio
+        :param addresses: An iterable of addresses to be included in the portfolio.
+        :type addresses: Iterable[Address]
+        """
         self._wallets: ChecksumAddressDict[PortfolioAddress] = ChecksumAddressDict({
+            """
+            :ivar _wallets: A checksummed dictionary of PortfolioAddress objects.
+            :vartype _wallets: ChecksumAddressDict[PortfolioAddress]
+            """
```
Comment (BobTheBuidler) [inline eth_portfolio/portfolio.py:60]: This should go on line 64

Comment (nvy-0x) [reply eth_portfolio/portfolio.py:60]: fixed

**Pull request merged**